src/btree/bt_cursor.c

```diff
@@ -575,9 +575,12 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
         }
     }
 
-    if (valid)
-        ret = __cursor_kv_return(cbt, cbt->upd_value);
-    else if (__cursor_fix_implicit(btree, cbt)) {
+    if (valid) {
```

Comment:
This is not correct. In the valid case, we have already done visibility check. I think we need to return before every time we call `__wt_cursor_valid `